processing/src/main/java/org/apache/carbondata/processing/merger/CarbonDataMergerUtil.java

```diff
@@ -1246,8 +1209,22 @@ public static boolean isHorizontalCompactionEnabled() {
     // set the update status.
     segmentUpdateStatusManager.setUpdateStatusDetails(segmentUpdateDetails);
 
-    CarbonFile[] deleteDeltaFiles =
-        segmentUpdateStatusManager.getDeleteDeltaFilesList(new Segment(seg), blockName);
+    // only when SegmentUpdateDetails contain the specified block
+    // will the method getDeleteDeltaFilesList be executed
+    List<String> blockNameList = segmentUpdateStatusManager.getBlockNameFromSegment(seg);
+    Map<String, List<CarbonFile>> blockAndDeleteDeltaFilesMap = new HashMap<>();
+    CarbonFile[] deleteDeltaFiles = null;
+    if (blockNameList.contains(blockName)) {
```

Comment:
if (blockNameList.contains(blockName)) { 
  blockAndDeleteDeltaFilesMap =
  if (blockAndDeleteDeltaFilesMap.containsKey(blockName)) {
  }
}

Reply:
Done. Combined the two judgement